### PR TITLE
Transfer image data from GCS to the site

### PR DESCRIPTION
**Pull request by thompsond**

### CODE REVIEW

File: src/main/java/com/google/step/QueryCloud.java [hunk 1]
```diff
@@ -11,15 +12,67 @@
...
+    private final String BUCKET_NAME = String.format("%s.appspot.com", PROJECT_ID);
+    private final static Logger LOGGER = Logger.getLogger(QueryCloud.class.getName());
+    private ArrayList<Item> items = null;
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // The request is made before the form is submitted (on page load)
+        if(items == null) {
```
Comment (gabbygermanson) [inline src/main/java/com/google/step/QueryCloud.java:34]: nit: space after if

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -1,5 +1,14 @@
 $(document).ready(function() {
- 
+    fetch("/query-cloud").then(res => res.json()).then(array => {
+        // Array is not equal to "{}" when the request is made after the form is submitted
+        if(array !== "{}"){
+            for(var i = 0; i < array.length; i++) {
+                const url = URL.createObjectURL(new Blob([Uint8Array.from(array[i].bytes).buffer], { type: "image/png" }));
+                $("#requestedImages").append(`<li><img src="${url}"></li>`);
```
Comment (gabbygermanson) [inline src/main/webapp/script.js:7]: Host note: This is where the hand-made image grid list would be developed with descriptions. (Until we may get the Google UI theme going).

File: src/main/webapp/script.js [hunk 3]
```diff
@@ -1,5 +1,14 @@
 $(document).ready(function() {
- 
+    fetch("/query-cloud").then(res => res.json()).then(array => {
+        // Array is not equal to "{}" when the request is made after the form is submitted
+        if(array !== "{}"){
```
Comment (gabbygermanson) [inline src/main/webapp/script.js:4]: nits: spacing after ifs and fors and before {

File: src/main/java/com/google/step/QueryCloud.java [hunk 4]
```diff
@@ -11,15 +12,67 @@
...
+        mapImages.forEach(image -> {
+            items.add(new Item(image, bucket.get(image.getObjectID()).getContent()));
+        });
+    }
+
+
+}
+
+// This class represents a MapImage and its corresponding image data
+class Item {
```
Comment (theannielin) [inline src/main/java/com/google/step/QueryCloud.java:66]: Use a more descriptive class name?

File: src/main/webapp/script.js (hunk 2 above)
Comment (theannielin) [reply src/main/webapp/script.js:7]: You can probs add a TODO for that here for now.

**Pull request merged**